tests/snappi_tests/intf_utils/intf_accept_metrics.py

```diff
@@ -0,0 +1,103 @@
+# This file defines the interfaces that snappi tests accept external metrics.
+import logging
+import json
+import datetime
+import time
+
+from typing import List, Dict, Union
+from intf_report_metrics import MetricReporterFactory OtelMetricReporter
+
+class Metric:
+    def __init__(self,
+                 name,
+                 description,
+                 unit,
+                 timestamp,
```

Comment:
The following fields is common for entire tests, so it can be move into the reporter as common metadata:

- testbed_name
- os_version
- testcase_name
- test_run_id


The following fields are common for all metrics in a single report action, so it can be lifted into the reporter's report function parameters:

- timestamp

The following fields are not clear on its purpose, we need to rename it to make it clear:

- component_id

Reply:
maybe the timestamp here means the test_start_time?